### PR TITLE
feature/create-dropdown-menu

### PR DESCRIPTION
**Pull request by Konmly**
Made a drop-down menu on the right, and the burger button made an animation on the cross.

https://user-images.githubusercontent.com/62384781/199106007-8cb3e424-4a1a-4615-93b3-32c72a43ac68.mp4

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -18,8 +18,17 @@
             <button type="button" class="header__button-outline button-outline">Регистрация</button>
         </div>
         <div class="header__mobile-button">
-          <button class="header__button-hamburger"></button>
-      </div>
+          <button class="header__button-hamburger">
+            <span class="header__button-cross"></span>
+            <nav class="header__mobile-menu">
```
Comment (PahaTonk) [inline src/index.html:23]: ??? what is that? Please read the docs
https://developer.mozilla.org/ru/docs/Web/HTML/Element/button#%D0%BE%D0%BF%D0%B8%D1%81%D0%B0%D0%BD%D0%B8%D0%B5
The button only supports text content

File: src/scss/_header.scss [hunk 2]
```diff
@@ -71,41 +71,94 @@
     display: none;
 
     @media screen and (max-width: $xs) {
+      display: flex;
```
Comment (PahaTonk) [inline src/scss/_header.scss:74]: you have a mixin for that

File: src/scss/_header.scss [hunk 3]
```diff
@@ -71,41 +71,94 @@
...
+        transition: 0.5s;
+
+        transform-origin: left;
+
+        background-color: $color__gray-99;
       }
 
-      &:hover {
-        display: none;
+      &:after {
```
Comment (PahaTonk) [inline src/scss/_header.scss:96]: Combine that rule with `&:before`

File: src/scss/_header.scss [hunk 4]
```diff
@@ -71,41 +71,94 @@
...
+        &::before {
+          transform: rotate(33deg);
+        }
+
+        &::after {
+          transform: rotate(-33deg);
+        }
+
+        span {
+          opacity: 0;
```
Comment (PahaTonk) [inline src/scss/_header.scss:118]: Does opacity appear smoothly?

File: src/scss/_header.scss [hunk 5]
```diff
@@ -71,41 +71,94 @@
...
+        }
       }
     }
   }
 
-  &__button {
```
Comment (PahaTonk) [inline src/scss/_header.scss:96]: what's about `header__button`?

File: src/scss/_header.scss [hunk 6]
```diff
@@ -71,41 +71,94 @@
...
+      &:focus {
+        &::before {
+          transform: rotate(33deg);
+        }
+
+        &::after {
+          transform: rotate(-33deg);
+        }
+
+        span {
```
Comment (PahaTonk) [inline src/scss/_header.scss:117]: don't use tag as selector

File: src/scss/_header.scss [hunk 7]
```diff
@@ -71,41 +71,94 @@
...
+      transform-origin: right;
+
+      background-color: $color__gray-99;
+    }
+  }
+
+  &__mobile-menu {
+    display: none;
+
+    & li {
```
Comment (PahaTonk) [inline src/scss/_header.scss:142]: as above

File: src/scss/_header.scss [hunk 8]
```diff
@@ -71,41 +71,94 @@
...
+      list-style: none;
     }
 
-    @media screen and (max-width: $lg) {
-      margin-right: 20px;
-      font-size: $font__size-14;
-      line-height: $line__height-16;
+    & a {
```
Comment (PahaTonk) [inline src/scss/_header.scss:146]: as above

File: src/scss/_header.scss [hunk 9]
```diff
@@ -71,41 +71,94 @@
...
+      text-decoration: none;
+    }
+
+    @media screen and (max-width: $xs) {
+      &:hover {
+        position: absolute;
```
Comment (PahaTonk) [inline src/scss/_header.scss:152]: Why is it needed?

File: src/scss/_header.scss [hunk 10]
```diff
@@ -71,41 +71,94 @@
...
+        display: flex;
```
Comment (PahaTonk) [inline src/scss/_header.scss:153]: use special mixin

**Review by PahaTonk — CHANGES_REQUESTED** [review]
Please record video how it work

File: src/scss/_header.scss (hunk 4 above)
Comment (Konmly) [reply src/scss/_header.scss:118]: yes

File: src/scss/_header.scss (hunk 5 above)
Comment (Konmly) [reply src/scss/_header.scss:96]: I can't find it in mine.

File: src/scss/_header.scss (hunk 6 above)
Comment (Konmly) [reply src/scss/_header.scss:117]: For some reason it doesn't want to work like this.
q1
q2

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/scss/_header.scss [hunk 11]
```diff
@@ -71,41 +71,88 @@
...
-      margin-right: 20px;
-      font-size: $font__size-14;
-      line-height: $line__height-16;
+  &__mobile-menu {
+    display: none;
+    transform: translateX(120%);
+
+    @media screen and (max-width: $xs) {
+      display: block;
+      transform: translateX(0);
```
Comment (PahaTonk) [inline src/scss/_header.scss:138]: I think that is needed when we have an `_active` modificator

File: src/index.html [hunk 12]
```diff
@@ -14,24 +14,34 @@
...
+            <span class="header__button-cross"></span>
+          </button>
         </div>
-        <div class="header__mobile-button">
-          <button class="header__button-hamburger"></button>
-      </div>
       </div>
     </header>
     <main>
+      <aside>
```
Comment (PahaTonk) [inline src/index.html:28]: [BEM] Element must have unique class

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 13]
```diff
@@ -14,24 +14,34 @@
...
+        <nav class="header__mobile-menu header__mobile-menu_active">
+          <ul class="header__mobile-list">
+            <li><a class="header__mobile-link" href="#">Bход</a></li>
```
Comment (PahaTonk) [inline src/index.html:31]: [BEM] Element must have unique class

File: src/index.html [hunk 14]
```diff
@@ -14,24 +14,34 @@
...
       <section class="octopus">
         <div class="octopus__container container">
           <div class="octopus__img">
             <img src="images/Octopus.png" alt="octopus picture" class="octopus__image" />
           </div>
           <h1 class="octopus__title">Обратный звонок на сайт</h1>
           <h3 class="octopus__subtitle">
-            Получайте в 3 раза больше звонков сразу после установки виджета на Ваш сайт Сервис
-            окупает свой годовой тариф в первую же неделю
+            <span>Получайте в 3 раза больше звонков сразу после установки виджета на Ваш сайт Сервис</span>
```
Comment (PahaTonk) [inline src/index.html:43]: [BEM] Element must have unique class

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/scss/_header.scss [hunk 15]
```diff
@@ -71,41 +71,100 @@
...
+        transition: 0.5s;
+
+        transform-origin: left;
+
+        background-color: $color__gray-99;
       }
 
-      &:hover {
-        display: none;
+      &:focus {
```
[the remark this answers is not in the document]
Comment (Konmly) [reply src/scss/_header.scss:95]: I couldn't find that line

q2

File: src/index.html [hunk 16]
```diff
@@ -14,33 +14,43 @@
...
         <div class="header__btns">
-            <button type="button" class="header__button button">Вход</button>
-            <button type="button" class="header__button-outline button-outline">Регистрация</button>
+          <button type="button" class="header__button button">Вход</button>
+          <button type="button" class="header__button-outline button-outline">Регистрация</button>
         </div>
         <div class="header__mobile-button">
-          <button class="header__button-hamburger"></button>
-      </div>
+          <button class="header__button-hamburger header__mobile-button_active">
```
Comment (PahaTonk) [inline src/index.html:21]: Why did you call `header__mobile-button_active`?

File: src/index.html [hunk 17]
```diff
@@ -14,33 +14,43 @@
...
+            <span class="header__button-cross"></span>
+          </button>
+        </div>
       </div>
     </header>
     <main>
+      <aside class="mobile__menu-wrap mobile__menu_active">
```
Comment (PahaTonk) [inline src/index.html:28]: Why did you add `-wrap` to the `mobile__menu-wrap` class ?

File: src/index.html [hunk 18]
```diff
@@ -14,33 +14,43 @@
...
+        <nav class="mobile__menu">
```
Comment (PahaTonk) [inline src/index.html:29]: It is not menu, it is navigation

File: src/index.html [hunk 19]
```diff
@@ -14,33 +14,43 @@
...
+          <ul class="mobile__list">
```
Comment (PahaTonk) [inline src/index.html:30]: where is the ELEMENT from the parent BLOCK?

File: src/index.html [hunk 20]
```diff
@@ -87,8 +97,8 @@ <h3 class="octopus__subtitle">
         <div class="diver__video video">
           <div class="video__container container">
             <h2 class="video__text">
-              <span>Посмотрите видео</span>
-              <span>как работает виджет</span>
+              <span class="video__text-watch">Посмотрите видео</span>
+              <span class="video__text-words">как работает виджет</span>
```
Comment (PahaTonk) [inline src/index.html:101]:
```suggestion
              <span class="video__text-description">Посмотрите видео</span>
              <span class="video__text-description">как работает виджет</span>
```

File: src/scss/_mobile-menu.scss [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+    position: fixed;
+    right: 0;
+    top: 70px;
+
+    width: 125px;
+
+    transition: 0.5s;
+
+    &:active //тут нужно будет поменять на класс active
+    {
```
Comment (PahaTonk) [inline src/scss/_mobile-menu.scss:18]: What is the problem? You can add the `active` modificator now

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/scss/_mobile-menu.scss [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+    width: 125px;
+
+    transition: 0.5s;
+
+    .sidebar__nav_active {
+      transform: translateX(0);
+    }
+
+    .sidebar__link {
+      display: block;
```
Comment (PahaTonk) [inline src/scss/_mobile-menu.scss:22]:
```suggestion
    .sidebar {
     &__nav_active {
      transform: translateX(0);
    }

    &__link {
      display: block;
```

File: src/scss/_mobile-menu.scss [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+
+      border-top-left-radius: 3px;
+      border-bottom-left-radius: 3px;
+
+      color: $color__white;
+      border-bottom: 1px solid $color__red-value;
+      background-color: $color__violet;
+
+      &:hover {
+        transition: 0.5s;
```
Comment (PahaTonk) [inline src/scss/_mobile-menu.scss:34]: move above

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/scss/_header.scss [hunk 24]
```diff
@@ -41,71 +41,74 @@
...
+  &__hamburger-wrap {
+    display: none;
 
     @media screen and (max-width: $xs) {
       display: block;
     }
   }
 
-  &__button-hamburger {
+  &__hamburger-button {
```
Comment (PahaTonk) [inline src/scss/_header.scss:52]: Where is that class in html?

File: src/index.html [hunk 25]
```diff
@@ -14,33 +14,43 @@
           <img src="images/Callmar.png" alt="picture of squid" />
         </div>
         <div class="header__btns">
-            <button type="button" class="header__button button">Вход</button>
-            <button type="button" class="header__button-outline button-outline">Регистрация</button>
+          <button type="button" class="header__button button">Вход</button>
+          <button type="button" class="header__button-outline button-outline">Регистрация</button>
+        </div>
+        <div class="header__hamburger-wrap header__hamburger_active">
```
Comment (PahaTonk) [inline src/index.html:20]: An element modifier cannot be above an element. It must be next to the element or the modifier is set to the wrong class

File: src/scss/_header.scss [hunk 26]
```diff
@@ -41,71 +41,74 @@
...
+
+        background-color: $color__gray-99;
       }
 
-      &:hover {
-        display: none;
+      .header__hamburger_active {
+        &::before {
+          transform: rotate(45deg);
+          width: 75%;
```
Comment (PahaTonk) [inline src/scss/_header.scss:80]: You can combine that rule with the rule on line 85

**Review by PahaTonk — CHANGES_REQUESTED** [review]

File: src/scss/_header.scss (hunk 24 above)
Comment (Konmly) [reply src/scss/_header.scss:52]: 21 line.
q1
